Automattic/jetpack
# Connection: Move the Jetpack IXR client to the package

In our quest to make the connection package fully independent, we still have to decouple some bits of it that belong to the Jetpack core. Some of those parts belong to other packages too, and some of them even belong to the connection package.

One of those pieces is the Jetpack IXR client. It is currently in the core of Jetpack, so we have to decide what to do with it. Since it relies on the connection package, and the connection package relies on it, that means that it makes sense to be part of the connection package.

So, this PR moves the Jetpack IXR client to the connection package.

#### Changes proposed in this Pull Request:
* Move `Jetpack_IXR_Client` to the connection package (classmapped).
* Move `Jetpack_IXR_ClientMulticall` to the connection package (classmapped).
* Introduce a basic `IXR` loader to load the WP core `IXR_` classes that we extend.
* Autoload the `IXR` loader by using the `files` Composer autoload directive.
* Deprecate `class.jetpack-ixr-client.php`.
* Deprecate `Jetpack::load_xml_rpc_client()`.
* Remove any unnecessary usage of `Jetpack::load_xml_rpc_client()`.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is a continuation of this effort: p1HpG7-7lI-p2

#### Testing instructions:
* 😅
* Checkout this branch.
* Run `composer install`
* Enable `WP_DEBUG` and `WP_DEBUG_LOG`.
* Test a bunch of things, like:
  * Connecting/reconnecting Jetpack to WP.com.
  * Disconnecting Jetpack from WP.com
  * Saving some Jetpack settings in wp-admin (toggles for example)
  * Saving some Jetpack settings in Calypso (toggles for example)
  * Incremental sync - trigger an incremental sync by creating a new draft post
  * Full sync - trigger a full sync from the debugger
  * Plugin/theme autoupdate
  * Optional: some Jetpack modules
    * Monitor - enable it and try breaking your site
    * Protect - enable it and try requesting the site from a forbidden IP
    * Post by email - try posting by email
    * Publicize - try it with a connected service and verify it publicized properly on it
    * Subscriptions - verify emails for new posts are still sent
    * SSO - try logging in with SSO.
* Verify there are no errors logged and everything worked well.
* Verify all tests still pass

#### Proposed changelog entry for your changes:
* Move `Jetpack_IXR_Client` to the connection package (classmapped).
* Move `Jetpack_IXR_ClientMulticall` to the connection package (classmapped).
* Introduce a basic `IXR` loader to load the WP core `IXR_` classes that we extend.
* Autoload the `IXR` loader by using the `files` Composer autoload directive.
* Deprecate `class.jetpack-ixr-client.php`.
* Deprecate `Jetpack::load_xml_rpc_client()`.
* Remove any unnecessary usage of `Jetpack::load_xml_rpc_client()`.
